cluster/cluster.go

```diff
@@ -759,11 +701,16 @@ func DeployNginxResolver() {
 		},
 	}
 
-	resDeployment, err := extV1beta1API(clientset).Deployments("default").Create(&deployment)
-	if err != nil {
-		panic(err.Error())
+	var resDeployment *v1beta1.Deployment
+	if shouldUpdate {
+		if resDeployment, err = extV1beta1API(clientset).Deployments("default").Update(&deployment); err != nil {
```

Comment:
We might need to check this since it seems that the Update is not being done.
My theory is that it might be because the deployment didn't change.